File: guestHouse-ui/src/services/ReservationService.js
import axios from "axios";

const BASE_REST_API_URL = 'http://localhost:8080/api/reservations';


export const getAllReservations = () => axios.get(BASE_REST_API_URL)

export const saveReservation = (reservation) => axios.post(BASE_REST_API_URL, reservation)

export const getReservation = (id) => axios.get(BASE_REST_API_URL + '/' + id)

export const updateReservation = (id, reservation) => axios.put(BASE_REST_API_URL + '/' + id, reservation)

export const deleteReservation = (id) => axios.delete(BASE_REST_API_URL + '/' + id)


  export const getAvailableDays = (startDate, endDate) => {
    return axios.get(`${BASE_REST_API_URL}/available-days?startDate=${startDate}&endDate=${endDate}`);
};  

File: guestHouse-management/src/main/java/net/javaguides/todo/service/ReservationService.java
package net.javaguides.todo.service;


import net.javaguides.todo.dto.ReservationDto;
import net.javaguides.todo.entity.Reservation;

import java.time.LocalDate;
import java.util.List;

public interface ReservationService {



    List<LocalDate> getAvailableDays(LocalDate startDate, LocalDate endDate);

    public List<ReservationDto> getAllReservations();

    public ReservationDto createReservation(ReservationDto reservationDto);

    ReservationDto getReservation(Long id);

    public void updateReservation(Long reservationId, Reservation updatedReservation);

    public void deleteReservation(Long id);


}
